src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderItemsController.cs

```diff
@@ -1,31 +1,76 @@
 using Microsoft.AspNetCore.Mvc;
+using NHSD.BuyingCatalogue.Ordering.Api.Extensions;
 using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
 using NHSD.BuyingCatalogue.Ordering.Common.Constants;
+using NHSD.BuyingCatalogue.Ordering.Domain;
 
 namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
 {
-    [Route("api/v1/orders/{orderId}")]
+    [Route("api/v1/orders/{orderId}/order-items")]
     [ApiController]
     [Produces(MediaTypeNames.Application.Json)]
     [Authorize(Policy = PolicyName.CanAccessOrders)]
     public sealed class OrderItemsController : ControllerBase
     {
+        private readonly IOrderRepository _orderRepository;
+
+        public OrderItemsController(IOrderRepository orderRepository)
+        {
+            _orderRepository = orderRepository ?? throw new ArgumentNullException(nameof(orderRepository));
+        }
+
         [HttpGet]
-        [Route("order-items")]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Swagger does not allow static files")]
-        public ActionResult<List<GetOrderItemModel>> Get(string orderId, [FromQuery] string catalogueItemType)
+        public async Task<ActionResult<IEnumerable<GetOrderItemModel>>> GetAllAsync(string orderId, [FromQuery] string catalogueItemType)
         {
-            if (orderId is null)
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+            if (order is null)
+            {
                 return NotFound();
+            }
+
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
+            {
+                return Forbid();
+            }
+
+            IEnumerable<OrderItem> orderItems = order.OrderItems;
+
+            if (!string.IsNullOrWhiteSpace(catalogueItemType))
+            {
+                var catalogueItemTypeFromName = CatalogueItemType.FromName(catalogueItemType);
+                if (catalogueItemTypeFromName is null)
+                {
+                    return new List<GetOrderItemModel>();
+                }
+
+                orderItems = orderItems.Where(y => y.CatalogueItemType.Equals(catalogueItemTypeFromName));
+            }
 
-            if (catalogueItemType is null)
-                throw new ArgumentNullException(nameof(catalogueItemType));
+            var serviceRecipients = order.ServiceRecipients;
 
-            return new List<GetOrderItemModel>();
+            return orderItems
+                .Select(orderItem => new GetOrderItemModel
+                {
+                    ItemId = $"{orderId}{orderItem.OdsCode}{orderItem.OrderItemId}",
+                    ServiceRecipient = new ServiceRecipientModel
+                    {
+                        Name = serviceRecipients.FirstOrDefault(serviceRecipient => string.Equals(orderItem.OdsCode,
```

Comment:
instead of linking like this all the time, why not use EF and have ServiceRecipient as a property?